ScienceDirect.js

```diff
@@ -48,26 +51,51 @@ function detectWeb(doc, url) {
 	}
 }
 
-function getPDFLink(doc) {
-	var pdfLink = ZU.xpathText(doc, '//div[@id="articleToolbar"]//a[@id="pdfLink" and not(@title="Purchase PDF")]/@href');
-	if (!pdfLink) {
-		pdfLink = ZU.xpathText(doc, '//div[@class="extendedPdfBox"]//a[@id="pdfLink" and not(@title="Purchase PDF")]/@href');
+function getPDFLink(doc, onDone) {
+	// No PDF access ("Get Full Text Elsewhere" or "Check for this article elsewhere")
+	if (doc.querySelector('.accessContent') || doc.querySelector('.access-options-link-text')) {
+		Zotero.debug("PDF is not available");
+		onDone();
+		return;
 	}
-	if (!pdfLink) {
-		pdfLink = ZU.xpathText(doc, '//div[@class="PdfEmbed"]/object/@data');
+	
+	// Some pages still have the PDF link available
+	var pdfURL = attr(doc, '#pdfLink', 'href');
+	if (pdfURL) {
+		onDone(pdfURL);
+		return;
```

Comment:
Though now I'm confused, because I'm seeing id="pdfLink" for all the links that open the menu. Was the initial commit of this actually working? I added the pdfLink part after getting the intermediate-page stuff working, so maybe I broke it after that? Anyway, seems OK now after ignoring '#'.